### PR TITLE
JAVA-5311 (take two)

### PR DESCRIPTION
**Pull request by jyemin**
Due to all the secrets still being echoed on the Gradle command line in the previous PR, this second attempt replaces all the Gradle system properties with environment variables.  As a result, a lot of code had to change but they were all fairly mechanical transformations (though not automated).

JAVA-5311

### CODE REVIEW

File: .evergreen/.evg.yml [hunk 1]
```diff
@@ -227,24 +227,29 @@ functions:
...
+          AZUREKMS_KEY_NAME: ${testazurekms_keyname}
         script: |
           ${PREPARE_SHELL}
-          export AWS_ACCESS_KEY_ID=${aws_access_key_id}
-          export AWS_SECRET_ACCESS_KEY=${aws_secret_access_key}
-          export AWS_DEFAULT_REGION=us-east-1
+          
           . ${DRIVERS_TOOLS}/.evergreen/csfle/set-temp-creds.sh
+          
+          export AWS_TEMP_ACCESS_KEY_ID=$CSFLE_AWS_TEMP_ACCESS_KEY_ID
```
Comment (jyemin) [inline .evergreen/.evg.yml:246]: These echo as

```
[2024/02/21 18:39:52.555] export AWS_TEMP_ACCESS_KEY_ID=$CSFLE_AWS_TEMP_ACCESS_KEY_ID
[2024/02/21 18:39:52.555] export AWS_TEMP_SECRET_ACCESS_KEY=$CSFLE_AWS_TEMP_SECRET_ACCESS_KEY
[2024/02/21 18:39:52.555] export AWS_TEMP_SESSION_TOKEN=$CSFLE_AWS_TEMP_SESSION_TOKEN
[2024/02/21 18:39:52.555] export CRYPT_SHARED_LIB_PATH=${CRYPT_SHARED_LIB_PATH}
```

Comment (jyemin) [reply .evergreen/.evg.yml:246]: @dylrich I'm not sure why it echoes that way, but is it expected and can I rely on it?

**Review by rozza — APPROVED** [review]
LGTM

**Review by dylrich — APPROVED** [review]
LGTM

**Pull request merged**